### ISSUE TITLE
Issues Importing or Updating dynatrace_hub_extension_config resource

### ISSUE THREAD

**Issue by jeremy-kaltenbach** [dynatrace-oss/terraform-provider-dynatrace#560]
**Describe the bug**
I have a SSL Certificate Monitor configured in Dynatrace Managed and am trying to import this existing resource over to Terraform. However, I am running into issues when it tries to apply the import (see screenshots below).

**To Reproduce**
In my Terraform module, I have this resource set up:
```
resource "dynatrace_hub_extension_config" "ssl_certificate_monitor" {
  count = var.environment == "test" || var.environment == "prod" ? 1 : 0
  name  = "com.dynatrace.custom.python-certificate-monitor"
  value = jsonencode(
    {
      "enabled" : true,
      "description" : "This belongs to DevOps",
      "version" : "1.8.13",
      "featureSets" : [
        "enhanced_metric"
      ],
      "activationContext" : "REMOTE",
      "pythonRemote" : {
        "alerting_configuration" : {
          "days_event_stage_2" : 5,
          "days_event_stage_1" : 10,
          "check_interval" : 1
        },
        "enable_ua_and_metrics" : {
          "enable_ua_and_metrics" : true
        },
        "log_event_interval" : {
          "log_event_interval_bool" : false
        },
        "check_hosts" : {
          "check_host_domain" : true,
          "domain_list" : var.dynatrace_ssl_host_domain[var.environment]
        },
        "debug" : true,
        "deep_debug" : null
      }
    }
  )
  active_gate_group = var.dynatrace_active_gate_group[var.environment]
}
```
and the following import:
```
import {
  for_each = var.environment == "test" || var.environment == "prod" ? { create = true } : {}
  to       = dynatrace_hub_extension_config.ssl_certificate_monitor[0]
  id       = "com.dynatrace.custom.python-certificate-monitor"
}
```
The `value` JSON snippet is pulled from the monitor configuration in Dynatrace Hub (with the `scope` configuration moved over to `active_gate_group`).

When running `terraform plan`, the `active_gate_group` is forcing Terraform to replace the resource:

![image](https://github.com/user-attachments/assets/1ee314ef-5b13-4b0c-94ea-aa982987ebe5)

And when running `terraform apply`,  I get a Constraints violated error saying the Version property is invalid, required format is: 1.0.0.

![image](https://github.com/user-attachments/assets/b55f66ba-1a37-4cf9-be16-ccb6ba195de0)

I then tried removing the `active_gate_group` configuration, so that Terraform doesn't delete/re-create the extension configuration. The execution plan then showed it as an update in place instead:
 
![image](https://github.com/user-attachments/assets/e01cd6fe-dc8b-4c9b-b845-b09c0b885bb8)

But applying produced a new error: "Not allow HTTP method, expected one of [POST, GET, OPTIONS]"

![image](https://github.com/user-attachments/assets/a2bd3d2e-a722-4128-a126-fbef852b9553)

This makes it sound like the Dynatrace provider does not allow updating or deleting the resource. I like to be able to just import the resource without Terraform trying to re-create or update it. Is that possible with the provider, or is there something wrong with my configuration? Or is there something else going on?

**Additional context**
This is using dynatrace-oss/dynatrace v1.62.0 with Terraform v1.9.3.
The access token used has the following scopes included: `extensions.read`, `extensions.write`, `extensionConfigurations.read`, `extensionConfigurations.write`.

**Comment by Reinhard-Pilz-Dynatrace** [comment]
Hi @jeremy-kaltenbach 

The reason for this weird behavior is the ID `com.dynatrace.custom.python-certificate-monitor` you're using within your `import` block

```
import {
  for_each = var.environment == "test" || var.environment == "prod" ? { create = true } : {}
  to       = dynatrace_hub_extension_config.ssl_certificate_monitor[0]
  id       = "com.dynatrace.custom.python-certificate-monitor"
}
```

The ID of a `dynatrace_hub_extension_config` is not **just** the ID of the extension. It is a concatenation of the extension ID and the ID of the monitoring configuration (usually a UUID). Something like this would be valid:

```
com.dynatrace.custom.python-certificate-monitor#-#2d7c9a5f-b2d5-3aa3-8921-91f8d9f7faac
```

That's an unfortunate consequence of how the REST API for maintaining extension works. Our Terraform Provider needs to know both identifiers in order to query for an Extension Configuration.

Via WebUI you can figure out that missing UUID by editing a Monitoring Configuration. The address line will contain that UUID.

![image](https://github.com/user-attachments/assets/a8e73df5-683b-419a-b1a7-7c652ae2ef74)

![image](https://github.com/user-attachments/assets/3b85d6de-dca3-4e09-b174-08bf07279c99)

Another, more convenient, way to get to the correct IDs of the currently configured `dynatrace_hub_extension_config`s on your environment would be to utilize the [Export functionality](https://docs.dynatrace.com/docs/manage/configuration-as-code/terraform/guides/export-utility) of the Terraform Provider.

You can read up on the full capabilities of that functionality in our documentation I've linked above. But in essence, for your specific use case, you just need to define credentials via the environment variables `DYNATRACE_ENV_URL` and `DYNATRACE_API_TOKEN` and then invoke the executable of the provider explicitly (instead as a plugin of Terraform).
#### Windows
```
terraform-provider-dynatrace_v1.62.0.exe -export -id dynatrace_hub_extension_config
```
#### Linux
```
terraform-provider-dynatrace_v1.62.0 -export -id dynatrace_hub_extension_config
```
The provider executable is usually available inside a subfolder `.terraform` of every Terraform module folder that is utilizing our provider.

The command line above produces a subfolder `configuration` and inside you will find readily available `.tf` files for every `dynatrace_hub_extension_config` found in your environment.

The `-id` flag ensures that every .`tf` file contains the ID of the setting as a comment on top.

```hcl
# ID com.dynatrace.custom.python-certificate-monitor#-#4b7e6745-c9a3-381c-a5de-5838b4159224
resource "dynatrace_hub_extension_config" "com_dynatrace_custom_python-certificate-monitor_1" {
  name              = "com.dynatrace.custom.python-certificate-monitor"
  active_gate_group = ....
}
```

I've noticed that your module adds an additional layer of complexity, utilizing `for_each` and `count` meta-attributes, that have a dependency on the variable `environment`. I'm not entirely sure whether I've grasped in full what your goal is here, but if there exists just a single `dynatrace_hub_extension_config` you would like to import, all you need to do is correcting the ID.

... except that you likely are still be getting these error message when running `terraform apply`.

```
╷
│ Error: DELETE https://######.live.dynatrace.com/api/v2/extensions/com.dynatrace.custom.python-certificate-monitor/monitoringConfigurations/: Constraints violated.
│ {"parameterLocation":"PATH","message":"Version property invalid. Required format is: 1.0.0","path":"extensionVersion"}
│
│
╵
```
Your state unfortunately already contains an instance of `dynatrace_hub_extension_config` with that incomplete ID. And Terraform would like to delete it. That entry within your state is pretty much the result of a failed import attempt.

Tomorrows release of the Terraform Provider will be able to handle that message for you. If you would like to move forward earlier you will have to remove that entry from your state manually.

Let me know if that helps,
Reinhard

**Comment by jeremy-kaltenbach** [comment]
Hi, @Reinhard-Pilz-Dynatrace 

Thanks for the response. I was able to delete the old resource from the state file and successfully import them with the correct IDs.
As for the `environment` variable I am using, I have the extension resource configured within one of our GitHub projects that has four deployment environments defined (test, stage, prod, and train). However, we only have two environments set up Dynatrace (non-prod, and prod). So the meta-attributes were used to make sure it only imports the two we have set up and not try to create extras.

To make sure it imported them both correctly, I added a new variable for the UUIDs of the monitoring configurations and then updated the import block to:
```
import {
  for_each = var.environment == "test" || var.environment == "prod" ? { create = true } : {}
  to       = dynatrace_hub_extension_config.ssl_certificate_monitor[0]
  id       = "com.dynatrace.custom.python-certificate-monitor#-#${var.dynatrace_ssl_extension_id[var.environment]}"
}
```
After applying the updates, the imports completed successfully with no forced replacements or in-place updates. Thanks again for your help!
